src/solver/ts-generator/generator.hxx

```diff
@@ -29,12 +29,52 @@
 
 #include <antares/logs/logs.h>
 
-namespace Antares
+namespace Antares::Solver::TSGenerator
 {
-namespace Solver
-{
-namespace TSGenerator
+ 
+inline void ResizeGeneratedTimeSeries(Data::AreaList& areas, Data::Parameters& params)
 {
+    for (auto i = areas.begin(); i != areas.end(); ++i)
+    {
+        auto& area = *(i->second);
+
+        // Load
+        if (params.timeSeriesToGenerate & Data::timeSeriesLoad)
+        {
+            area.load.series.timeSeries.reset(params.nbTimeSeriesLoad, HOURS_PER_YEAR);
+        }
+
+        // Wind
+        if (params.timeSeriesToGenerate & Data::timeSeriesWind)
+        {
+            area.wind.series.timeSeries.reset(params.nbTimeSeriesWind, HOURS_PER_YEAR);
+        }
+
+        // Solar
+        if (params.timeSeriesToGenerate & Data::timeSeriesSolar)
+        {
+            area.solar.series.timeSeries.reset(params.nbTimeSeriesSolar, HOURS_PER_YEAR);
+        }
+
+        // Hydro
+        if (params.timeSeriesToGenerate & Data::timeSeriesHydro)
+        {
+            area.hydro.series->resizeRORandSTORAGE(params.nbTimeSeriesHydro);
+        }
+
+        // Thermal
+        auto end = area.thermal.list.mapping.end();
+        for (auto it = area.thermal.list.mapping.begin(); it != end; ++it)
+        {
+            auto& cluster = *(it->second);
+            bool globalThermalTSgeneration = params.timeSeriesToGenerate & Data::timeSeriesThermal;
```

Comment:
This constant can go outside the loop

Reply:
Done (caution : due to another remark from @sylvlecl, this loop, and its context, was moved to a file **generator.cpp**)